addons/binding/org.openhab.binding.roku/src/main/java/org/openhab/binding/roku/internal/protocol/RokuCommunication.java

```diff
@@ -0,0 +1,209 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.roku.internal.protocol;
+
+import static org.openhab.binding.roku.RokuBindingConstants.*;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.StringReader;
+import java.lang.reflect.Field;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLConnection;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+
+import org.apache.commons.io.IOUtils;
+import org.eclipse.smarthome.core.library.types.RawType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.openhab.binding.roku.internal.RokuState;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.InputSource;
+
+/**
+ * The {@link RokuCommunication} class is what communicates directly with a Roku device (sending commands to it,
+ * updating state, etc).
+ *
+ * @author Jarod Peters - Initial contribution
+ */
+public class RokuCommunication {
+    private final Logger logger = LoggerFactory.getLogger(RokuCommunication.class);
+    private final DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+    private final String host;
+    private final Number port;
+
+    public RokuCommunication(String host, Number port) {
+        this.host = host;
+        this.port = port;
+    }
+
+    public String getHost() {
+        return host;
+    }
+
+    public Number getPort() {
+        return port;
+    }
+
+    private StringType getTagName(String tag, Element eElement) {
+        return new StringType(eElement.getElementsByTagName(tag).item(0).getTextContent());
+    }
+
+    private String getSubTagName(String tag, Element eElement) {
+        return eElement.getElementsByTagName(tag).item(0).getAttributes().getNamedItem("id").toString()
+                .replace("id=", "").replace("\"", "");
+    }
+
+    /**
+     * Future Functionality
+     * private HashMap<String, String> getAllSubTags(String tag, Element eElement) {
```

Comment:
Please remove this commented out code.

_if code is refactored in the meantime its unlikely that the commented out code is also modified, so as soon as the code might be needed it again it might not even work anymore_

_http://softwareengineering.stackexchange.com/questions/190096/can-commented-out-code-be-valuable-documentation_